### PR TITLE
Add listProcOn feature

### PR DESCRIPTION
**Pull request by jbkim1999**
Along with its parser, and comparators for the ease of comparing between properties of different Procedures.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ListProcOnCommand.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        Collections.sort(procedureClientPairList, new Comparator<Pair<Procedure, Client>>() {
+            @Override
+            public int compare(Pair<Procedure, Client> p1, Pair<Procedure, Client> p2) {
+                return p1.getKey().compareTo(p2.getKey());
+            }
+        });
+
+        for (int i = 0; i < procedureClientPairList.size(); i++) {
```
Comment (jbkim1999) [inline src/main/java/seedu/address/logic/commands/ListProcOnCommand.java:60]: Not really sure whether formatting should be done in this method as well.
Are there are methods that are better fit to handle the formatting after passing down the `CommandResult`?

File: src/main/java/seedu/address/logic/commands/ListProcOnCommand.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            clientProceduresOnDate.stream().map(Procedure -> new Pair(Procedure, currentClient))
+                    .forEach(procedureClientPairList::add);
+        }
+
+        Collections.sort(procedureClientPairList, new Comparator<Pair<Procedure, Client>>() {
+            @Override
+            public int compare(Pair<Procedure, Client> p1, Pair<Procedure, Client> p2) {
+                return p1.getKey().compareTo(p2.getKey());
+            }
+        });
```
Comment (iztanpy) [inline src/main/java/seedu/address/logic/commands/ListProcOnCommand.java:58]: would it be better to make this method outside of the execute method?

File: src/main/java/seedu/address/logic/commands/ListProcOnCommand.java (hunk 1 above)
Comment (iztanpy) [reply src/main/java/seedu/address/logic/commands/ListProcOnCommand.java:60]: maybe can try to abstract out the formatting to take just index, client and procedure and then get the data out from there? Then maybe we can call that method in the execute command

**Review by aceszhenwei — COMMENTED** [review]
Nicely done! I don't see any serious issues. As @iztanpy mentioned, there is a possibility of abstracting out the formatting instead. Apart from that, if you want, you can also to break down the the `Collections.sort` to avoid any confusion.

**Pull request merged**